packages/back-end/src/controllers/experiments.ts

```diff
@@ -2130,10 +2130,7 @@ export async function postPastExperiments(
   if (!datasourceObj) {
     throw new Error("Could not find datasource");
   }
-  req.checkPermissions(
-    "runQueries",
-    datasourceObj?.projects?.length ? datasourceObj.projects : ""
-  );
+  req.checkPermissions("runQueries", datasourceObj.projects);
```

Comment:
For resources that have a projects array, passing in an empty array indicates that it's a part of "ALL PROJECTS". We should only pass in an empty string if we want to check the user's global permissions, or for resources that have a single `project` field like features and experiments.